filebeat/input/filestream/internal/input-logfile/metrics.go

```diff
@@ -63,11 +63,7 @@ type Metrics struct {
 }
 
 func NewMetrics(reg *monitoring.Registry) *Metrics {
-	// The log input creates the `filebeat.harvester` registry as a package
-	// variable, so it should always exist before this function runs.
-	// However, at least on testing scenarios this does not hold true, so
-	// if needed, we create the registry ourselves.
-	harvesterMetrics := monitoring.Default.GetOrCreateRegistry("filebeat.harvester")
+	harvesterMetrics := reg.GetOrCreateRegistry("filebeat.harvester")
```

Comment:
NIT: Any way we can write a test that previously reproduced the panic but now passes?